main.go

```diff
@@ -109,6 +109,16 @@ func (c *gandiDNSProviderSolver) Present(ch *v1alpha1.ChallengeRequest) error {
 		entry = strings.TrimSuffix(entry, ".")
 		entry = strings.Replace(entry, ".", "_", -1)
 		entry = entry + ".verify"
+
```

Comment:
Reading the code for the webhook, I think maybe the only bug is that we are using `cn.ResolvedFQDN` instead of `domain` which is returned from `c.getDomainAndEntry()`? That function seems to strip both the `.` and the `cn.ResolvedZone` which (probably) would be the equivalent of at least `molnett.net` in our case.

Then we would not need the nested `strings.Replace` that you have - but I may be reading the code wrong!